engine/verification/test/helper.go

```diff
@@ -505,3 +505,53 @@ func IsAssigned(index uint64, chunkNum int) bool {
 func isSystemChunk(index uint64, chunkNum int) bool {
 	return int(index) == chunkNum-1
 }
+
+func CreateExecutionResult(blockID flow.Identifier, options ...func(result *flow.ExecutionResult, assignments *chunks.Assignment)) (*flow.ExecutionResult, *chunks.Assignment) {
+	result := &flow.ExecutionResult{
+		BlockID: blockID,
+		Chunks:  flow.ChunkList{},
+	}
+	assignments := chunks.NewAssignment()
+
+	for _, option := range options {
+		option(result, assignments)
+	}
+	return result, assignments
+}
+
+func WithChunks(setAssignees ...func(flow.Identifier, uint64, *chunks.Assignment) *flow.Chunk) func(*flow.ExecutionResult, *chunks.Assignment) {
+	return func(result *flow.ExecutionResult, assignment *chunks.Assignment) {
+		for i, setAssignee := range setAssignees {
+			chunk := setAssignee(result.BlockID, uint64(i), assignment)
+			result.Chunks.Insert(chunk)
+		}
+	}
+}
+
+func ChunkWithIndex(blockID flow.Identifier, index int) *flow.Chunk {
+	chunk := &flow.Chunk{
+		Index: uint64(index),
+		ChunkBody: flow.ChunkBody{
+			CollectionIndex: uint(index),
+			EventCollection: blockID, // ensure chunks from different blocks with the same index will have different chunk ID
+			BlockID:         blockID,
+		},
+		EndState: unittest.StateCommitmentFixture(),
+	}
+	return chunk
+}
+
+func WithAssignee(assignee flow.Identifier) func(flow.Identifier, uint64, *chunks.Assignment) *flow.Chunk {
+	return func(blockID flow.Identifier, index uint64, assignment *chunks.Assignment) *flow.Chunk {
+		chunk := ChunkWithIndex(blockID, int(index))
+		fmt.Printf("with assignee: %v, chunk id: %v\n", index, chunk.ID())
+		assignment.Add(chunk, flow.IdentifierList{assignee})
+		return chunk
+	}
+}
+
+func FromChunkID(chunkID flow.Identifier) flow.ChunkDataPack {
+	return flow.ChunkDataPack{
+		ChunkID: chunkID,
+	}
+}
```

Comment:
These are basically what we moved from `match` to this test `helper` so that we can share it among different packages (**nothing new to review**).